src/Compilers/Core/Portable/Operations/OperationCache.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Immutable;
+
+namespace Microsoft.CodeAnalysis
+{
+    /// <summary>
+    /// Cache data associated with TBoundNode
+    /// </summary>
+    internal class OperationCache<TBoundNode>
+    {
+        // REVIEW: not sure about perf of value tuple used as key on a dictionary. I might need custom type
```

Comment:
Code in feature branches should not add "REVIEW" and other "TODO" markers. You can either (1) use "PROTOTYPE(<feature name>)" instead or (2) file a tracking issue on github.
Note that all PROTOTYPE markers need to be resolved (or converted to issues) before a feature can be merged to master branch.